extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java

```diff
@@ -1174,24 +1186,103 @@ private static void collectDialectConfig(String persistenceUnitName,
                     persistenceUnitName));
         }
 
-        if (persistenceUnitConfig.dialect().storageEngine().isPresent()) {
-            // Only actually set the storage engines if MySQL or MariaDB
-            if (isMySQLOrMariaDB(dbKind, dialect)) {
-                // The storage engine has to be set as a system property.
-                // We record it so that we can later run checks (because we can only set a single value)
-                storageEngineCollector.add(persistenceUnitConfig.dialect().storageEngine().get());
-                systemProperties.produce(new SystemPropertyBuildItem(AvailableSettings.STORAGE_ENGINE,
-                        persistenceUnitConfig.dialect().storageEngine().get()));
-            } else {
+        if (dbProductVersion.isPresent()) {
+            puPropertiesCollector.accept(AvailableSettings.JAKARTA_HBM2DDL_DB_VERSION, dbProductVersion.get());
+        }
+
+        handleDialectSpecificSettings(
+                persistenceUnitName,
+                systemProperties,
+                puPropertiesCollector,
+                storageEngineCollector,
+                dialectConfig,
+                dbKind,
+                dialect);
+    }
+
+    private static void handleDialectSpecificSettings(
+            String persistenceUnitName,
+            BuildProducer<SystemPropertyBuildItem> systemProperties,
+            BiConsumer<String, String> puPropertiesCollector,
+            Set<String> storageEngineCollector,
+            HibernateOrmConfigPersistenceUnit.HibernateOrmConfigPersistenceUnitDialect dialectConfig,
+            Optional<String> dbKind,
+            Optional<String> dialect) {
+
+        // todo : do we want dbKind/dialect aware handling for all of these settings (i.e. isMySQLOrMariaDB)?
+        //      if so, we need to reorganize this code a bit
```

Comment:
When we know for sure what the db-kind/dialect is, indeed it would probably be a good idea to at least warn users if they set dialect options that are not going to have any effect -- like we do for `storage-engine`. We can even consider throwing an exception, but will need to mention this in the migration guide.